codegen/gce/src/lib.rs

```diff
@@ -0,0 +1,248 @@
+use ir::{
+    constraints::{ConstantValue, Operation},
+    AirIR,
+};
+
+pub use air_script_core::{
+    Constant, ConstantType, Expression, Identifier, IndexedTraceAccess, MatrixAccess,
+    NamedTraceAccess, TraceSegment, Variable, VariableType, VectorAccess,
+};
+use std::fmt::Display;
+
+mod error;
+use error::ConstraintEvaluationError;
+
+mod utils;
+
+mod expressions;
+use expressions::ExpressionsHandler;
+
+use std::collections::BTreeMap;
+use std::fs::File;
+use std::io::Write;
+
+/// Holds data for JSON generation
+#[derive(Default, Debug)]
+pub struct CodeGenerator {
+    num_polys: u16,
+    num_variables: usize,
+    constants: Vec<u64>,
+    expressions: Vec<ExpressionJson>,
+    outputs: Vec<usize>,
+}
+
+impl CodeGenerator {
+    pub fn new(ir: &AirIR, extension_degree: u8) -> Result<Self, ConstraintEvaluationError> {
+        // maps indexes in Node vector in AlgebraicGraph and in `expressions` JSON array
+        let mut expressions_map = BTreeMap::new();
```

Comment:
This isn't used at this level, so it can be removed here and created directly when the `ExpressionsHandler` is instantiated